### ISSUE TITLE
define new Region Of interest after all tracks have been rendered

### ISSUE THREAD

**Issue by abhinav-strand** [igvteam/igv.js#996]
How to define new Region Of interest on the fly after all tracks have been rendered(without re-rendering the browser)

**Comment by jrobinso** [comment]
I'm not sure if this is a question or a feature request, but you can't do that.   If you would like to request it please provide more details about your use case and how it would work.   Define a ROI through the user interface, or in javascript code?

**Comment by abhinav-strand** [comment]
Thanks for replying. Actually it is a question for now. But if not possible under current igv version, then it should be considered as feature request.

The feature is as such-

The user gets an input ui with chromosome, start and end fields. User enters the value and wants to create a ROI out of it on igv browser. This happens after igv browser rendering with all the tracks has happened.
So, is there a way to define ROI on the fly without reloading the browser?

**Comment by jrobinso** [comment]
No there is not currently.

**Comment by abhinav-strand** [comment]
Can it be taken as a feature request?

**Comment by jrobinso** [comment]
Yes, but in the scenario you describe the ROI will be lost when the user closes the browser or reloads the page.

**Comment by abhinav-strand** [comment]
That is fine. The requirement is to have to a temporary ROI created on the fly without the need to call igv.createBrowser() method

**Comment by jrobinso** [comment]
@blajoie we looked at that initially but it turns out to be a logistical bookkeeping headache with no real benefit.   Unlike the sweeper in the ruler the regions have genomic coordinates, they need to pan left and right and zoom in and out with everything else.   Also they need exported to svg and png images like everything else.      Painting is very fast, and an image tile is only occasionally painted.

What you suggest is probably the best approach for defining a ROI however.  Once the mouse is released an ROI can be created and treated as currently.   

ROIs can be general or track specific so that needs to be considered, perhaps the dynamic ones are all general.

**Comment by pratibha-strand** [comment]
@jrobinso - Would it be possible to add a simple API to add ROI something like addTrack, which will take care of updating browser ROI and updates the track views? This would solve our problem with minimal changes.

**Comment by jrobinso** [comment]
@pratibha-strand Brilliantly simple solution.   I just pushed this, the function is ```loadROI```, it will take a single ROI config or an array.  See example below.  
```
browser.loadROI([
                        {
                            name: 'ROI set 1',
                            url: 'https://s3.amazonaws.com/igv.org.test/data/roi/roi_bed_1.bed',
                            indexed: false,
                            color: "rgba(68, 134, 247, 0.25)"
                        },
                        {
                            name: 'ROI set 2',
                            url: 'https://s3.amazonaws.com/igv.org.test/data/roi/roi_bed_2.bed',
                            indexed: false,
                            color: "rgba(0, 150, 50, 0.25)"
                        }
```

 BTW the api function is ```loadTrack```,  ```addTrack``` is not part of the API, its name could change or be refactored out altogether at any time.  The API are the functions defined in the wiki.

**Comment by pratibha-strand** [comment]
@jrobinso Thanks for your quick fix for this. I understand that the `addTrack` is not part of the API. Then there is no API to remove track or remove ROI. Isn't it equally important to be able to load and remove tracks/ROI from the browser? 

Can you please expose an API to do the same.

**Comment by jrobinso** [comment]
Yes, good point, can you open a new issue for that?  Thanks.

**Comment by jrobinso** [comment]
Closing this as loadROI is released with 2.3.5.   "removeROI" is covered by #1013
